spatialos-sdk/src/worker/entity.rs

```diff
@@ -0,0 +1,71 @@
+use crate::worker::component::{self, Component, ComponentId};
+use spatialos_sdk_sys::worker::Worker_ComponentData;
+use std::collections::HashMap;
+use std::ptr;
+
+#[derive(Debug)]
+struct ComponentData {
+    raw_data: Worker_ComponentData,
```

Comment:
As mentioned in my other comment, I'm quite against having the `Entity` having any knowledge of `Worker_ComponentData` and instead storing components inside a type erased `Box<Component>` instead, and copy the data later into an `Arc<Component>` later. Not sure how easy that is to accomplish if everything is typed erased though. Might need to rely on the component database passed into the connection object in some way.

Reply:
Hrmmm, there are slight performance differences between the two approaches, but I'd expect them to be negligible. Really, storing the `Box<Component>` is likely more efficient because we aren't creating an `Arc<T>`, and dealing with the atomic refcount is likely the biggest performance difference between the two approaches.

I think we're imagining different purposes for `Entity`, though. I wasn't expecting it to be a general-purpose view into an entity's data, I was only planning on using it purely for the purpose of gathering data to send to the C API (hence why I'm inclined to just use `Worker_ComponentData` internally). I'm not yet sure what a more general-purpose component data API would look like at this point, I'll likely have to see what use cases I run into as I work more with SpatialOS.